xmtp_mls/src/client.rs

```diff
@@ -107,23 +107,24 @@ where
         created_before_ns: Option<i64>,
         limit: Option<i64>,
     ) -> Result<Vec<MlsGroup<ApiClient>>, ClientError> {
-        Ok(self
-            .store
-            .find_groups(
-                &mut self.store.conn()?,
-                allowed_states,
-                created_after_ns,
-                created_before_ns,
-                limit,
-            )?
-            .into_iter()
-            .map(|stored_group| MlsGroup::new(self, stored_group.id, stored_group.created_at_ns))
-            .collect())
+        Ok(EncryptedMessageStore::find_groups(
```

Comment:
I'm perfectly fine with making all the store methods static, given how few of them ever need to call another store method. 

There's probably a case to make that these should just be methods on the models `Groups::find` instead of `EncryptedMessageStore::find_groups`. Probably more a matter of taste. Having them all be methods on one big store does limit the number of imports needed, which is simpler.

Good to leave as is setup in this PR if everyone else feels the same way.

Reply:
I don't have a problem reverting back if there are strong feelings around it, but it does clean up some of the fields in structs as well

I'm not sure how much more the store will grow, but considering we already have most of the methods within their model files, it shouldn't be difficult to refactor if we decide that pulling in all the methods is too much down the line

Reply:
Good to leave it for now, and we can re-open this discussion if the `EncryptedMessageStore` keeps growing out of control and becomes unwieldy.

Reply:
I was actually wondering if these made sense as non-static methods on the connection. Every DB operation requires a connection anyway. But happy with this too

Reply:
That's what Wire does and the ergonomics actually seem pretty nice to me

Reply:
Essentially we would then just be wrapping diesel in a NewType and abstracting it away which I would generally be in favor of

Reply:
I can exp with this tonight